Add funding.json and .well-known/funding-manifest-urls

Robot8A opened the pull request:
Added [funding.json](https://fundingjson.org/) and well-known files, for describing OSM's fundraising options.
This would allow OSM to apply for [FLOSS/fund](https://floss.fund/) funding.

I added OSM (database) as the sole project. An alternative arrangement would be to describe the different software projects that form the OSM Core software, with its associated licenses.

tomhughes commented:
Thinking about it this will probably have to be done via https://github.com/openstreetmap/chef/ and not here as we don't want other people running this code to be publishing this without realising.

Frankly the whole thing of publishing a JSON document nobody will ever read seems weird and I have to wonder about an organisation that requires it to consider applications...

Robot8A commented:
I made some fixes to pass the https://dir.floss.fund/validate validation tool.
These are:
- Change descriptions to less than 500 characters
- Have well-known URLs have the same hostname as the project/organization/repo URL

As well, I fixed:
- License namespace
- Added GH Sponsorship

pnorman commented on public/funding.json:
```diff
@@ -21,8 +21,8 @@
-        "url": "https://planet.openstreetmap.org",
-        "wellKnown": "https://planet.openstreetmap.org/.well-known/funding-manifest-urls"
+        "url": "https://github.com/openstreetmap/",
+        "wellKnown": "https://github.com/openstreetmap/chef/tree/master/cookbooks/web/files/default/static/.well-known/funding-manifest-urls"
```
We should be hosting this file, not relying on github. Additionally, it's unlikely that this URL will meet the format required, as it's not the URL of the file, it's a URL for a web page showing the github interface.

Robot8A commented:
https://floss.fund/blog/second-tranche-2025-anniversary/
